app/src/main/java/unipd/se18/ocrcamera/PhotoDownloadTask.java

```diff
@@ -0,0 +1,216 @@
+package unipd.se18.ocrcamera;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.os.Environment;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.ScrollView;
+import android.widget.TextView;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.commons.net.ftp.FTPReply;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+/**
+ * @author Leonardo Rossi (g2) and Stefano Romanello (g3)
+ */
+public class PhotoDownloadTask extends AsyncTask<Void, Integer, Void>
+{
+
+    private FTPClient ftp;
+    //private ArrayList<String> messages;
+    private Context context;
+    private ProgressBar progressBar;
+    private TextView textViewProgress;
+    private Button button;
+    private Integer currentProgress;
+    private ScrollView scrollCurrentDownload;
+    private TextView txtViewCurrentDownload;
+    //Constants
+    // PLEASE DON'T USE THESE FOR PERSONAL USAGE!
```

Comment:
Could public the credentials here be danger? Providing an activity that asks the credentials could be more secure?

Reply:
No, they're danger if you use them for other purposes.

Reply:
Could you restrict this user? Is it a good idea? This repo is public, is it dangerous having the credentials in the code or it could be better to ask to the user these credentials?

Reply:
I wanted to create other credentials but the free hosting doesn't let me do that

Reply:
Ok, could be a good idea asking to the user these credentials instead writing it in this public code?